tests/test_sourmash.py

```diff
@@ -2832,6 +2832,218 @@ def test_multigather_metagenome_query_from_file(c):
+
+    cmd = 'multigather --query-from-file gcf_all.sbt.zip --db gcf_all.sbt.zip -k 21 --threshold-bp=0'
+    cmd = cmd.split(' ')
+    c.run_sourmash(*cmd)
+
+    out = c.last_result.out
+    print(out)
+    err = c.last_result.err
+    print(err)
+
+    assert 'conducted gather searches on 12 signatures' in err
+    assert 'the recovered matches hit 100.0% of the query' in out
+    assert all(('4.7 Mbp      100.0%  100.0%'  in out,
+                'NC_011080.1 Salmonella enterica subsp...' in out))
+    assert all(('4.5 Mbp      100.0%  100.0%' in out,
+                'NC_004631.1 Salmonella enterica subsp...' in out))
+
+
+@utils.in_tempdir
+def test_multigather_metagenome_sbt_query_from_file_with_addl_query(c):
+
+    testdata_glob = utils.get_test_data('gather/GCF*.sig')
+    testdata_sigs = glob.glob(testdata_glob)
+
+    query_sig = utils.get_test_data('gather/combined.sig')
+
+    cmd = ['index', 'gcf_all.sbt.zip', '-k', '21']
+    cmd.extend(testdata_sigs)
+    c.run_sourmash(*cmd)
+
+    assert os.path.exists(c.output('gcf_all.sbt.zip'))
+
+    another_query = utils.get_test_data('gather/GCF_000195995.1_ASM19599v1_genomic.fna.gz.sig')
+
+    cmd = 'multigather --query {} --query-from-file gcf_all.sbt.zip --db gcf_all.sbt.zip -k 21 --threshold-bp=0'.format(another_query)
+    cmd = cmd.split(' ')
+    c.run_sourmash(*cmd)
+
+    out = c.last_result.out
+    print(out)
+    err = c.last_result.err
+    print(err)
+
+    assert 'conducted gather searches on 13 signatures' in err
+    assert 'the recovered matches hit 100.0% of the query' in out
+    assert all(('4.7 Mbp      100.0%  100.0%'  in out,
+                'NC_011080.1 Salmonella enterica subsp...' in out))
+    assert all(('4.5 Mbp      100.0%  100.0%' in out,
+                'NC_004631.1 Salmonella enterica subsp...' in out))
+
+
+@utils.in_tempdir
+def test_multigather_metagenome_lca_query_from_file(c):
+    testdata_glob = utils.get_test_data('47*.fa.sig')
+    testdata_sigs = glob.glob(testdata_glob)
+    #testdata_sig1 = utils.get_test_data('47.fa.sig')
+    #testdata_sig2 = utils.get_test_data('63.fa.sig')
+
+    lca_db = utils.get_test_data('lca/47+63.lca.json')
+
+    cmd = ['index', '47+63.sbt.zip', '-k', '31']
+    cmd.extend(testdata_sigs)
+    c.run_sourmash(*cmd)
+
+    assert os.path.exists(c.output('47+63.sbt.zip'))
+
+    # currently failing bc lca json gets read as signatures (line 719 causes failure line 726 in commands.py),
+    cmd = 'multigather --query-from-file {} --db 47+63.sbt.zip -k 31 --threshold-bp=0'.format(lca_db)
+    cmd = cmd.split(' ')
+    c.run_sourmash(*cmd)
+
+    out = c.last_result.out
+    print(out)
+    err = c.last_result.err
+    print(err)
+
+    assert 'conducted gather searches on 2 signatures' in err
+    assert 'the recovered matches hit 100.0% of the query' in out
+    assert '5.1 Mbp      100.0%   64.9%    491c0a81'  in out
+    assert '5.5 Mbp      100.0%   69.4%    491c0a81'  in out
+
+
```

Comment:
@ctb this test is failing bc line 719 reads in the lca json file --> signatures, rather than just adding it to the `inp_files` list. Advice on how to handle?

Reply:
right! it should be `--query {}` isntead of `--query-from-file`.

The other tests we're missing (this is not necessarily on you to add, tho!) are ones where we put the name of an LCA and/or SBT file in `query.list` and pass it in with `--query-from-file`.

Reply:
ah. This makes sense -- for some reason I thought you meant `query-from-file` should also accept db's.